src/js/tasks/osSiloPrice.js

```diff
@@ -34,7 +37,9 @@ const setOSSiloPrice = async (options) => {
     log(`Current market pricing: ${Number(formatUnits(currentPricing, 18)).toFixed(4)}`);
 
     // 4. Calculate highest buy price, we should always target a price lower than this to maintain the APY
-    const minBuyingPrice = calculateMinBuyingPrice(currentApyLending);
+    const duration = await estimateAverageWithdrawTime(arm, block);
+    //const duration = await calculateAveragePeriod(arm);
+    const minBuyingPrice = calculateMinBuyingPrice(currentApyLending, duration);
```

Comment:
We get the `currentApyLending` from lending market and `currentPricing` from fly.trade (used in the further computations). Both of these fetch the latest price, not historical price if a specific `block` is used. Should we do some sort of preventive measures to avoid such inconsistencies?

Reply:
I used the `block` mostly to "test" the behavior of the code and the calculation of the withdrawal period, without having any intention of using it in production.
Do you think logging a message is enough? Or would you prefer to throw an error or something?

Reply:
If it isn't gonna be used on production, just throwing an error when it's deployed would be fine. But works either way